### PR TITLE
samples: drivers: i2c: I2C scanner

### PR DESCRIPTION
**Pull request by ntavish**
Scans i2c bus for any devices present.

### CODE REVIEW

**Review by pabigot — CHANGES_REQUESTED** [review]
Other than the issues listed in the review I believe standard practice requires a `sample.yaml` file so this gets tested, but I'm not sure how strict that is.

I've confirmed this works on an nrf51_ble400 with four I2C devices.  It does not detect the I2C devices on the nrf52_pca20020 but I'm not sure that's a problem with this code.  (The I2C pins on that board are not accessible so I can't see what's going on.)

**Comment by ntavish** [comment]
I've added a `sample.yaml`, but not sure how it would do the test..

**Comment by ntavish** [comment]
@nashif I'll take a look into how to add commands to the shell when I get a chance, this would be useful probably.

**Review by pabigot — APPROVED** [review]

**Comment by pabigot** [comment]
> I've added a `sample.yaml`, but not sure how it would do the test..

Neither do I, but it's there--thanks.  Another reviewer will, hopefully, have more experience with the Zephyr infrastructure and be able to validate it.

File: samples/drivers/i2c_scanner/README.rst [hunk 1]
```diff
@@ -0,0 +1,29 @@
+.. _i2c_scanner:
+
+I2C Scanner sample
+##################
+
+Overview
+********
+This sample sends I2C messages without any data (i.e. stop condition
+after sending the just the address), and if there is an ACK for the
```
Comment (dbkinder) [inline samples/drivers/i2c_scanner/README.rst]:
```suggestion
after sending just the address). If there is an ACK for the
```

File: samples/drivers/i2c_scanner/README.rst [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+address, it prints it prints the address as ``FOUND``.
```
Comment (dbkinder) [inline samples/drivers/i2c_scanner/README.rst]:
```suggestion
address, it prints the address as ``FOUND``.
```

File: samples/drivers/i2c_scanner/README.rst [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+Warning!
```
Comment (dbkinder) [inline samples/drivers/i2c_scanner/README.rst]: Oh dear.  Let's make this more prominent:

```
.. warning:: As  there  is  no  standard I2C detection command, this sample
   uses arbitrary SMBus commands (namely SMBus quick write and SMBus
   receive byte) to probe for devices.  This sample program can confuse
   your I2C bus, cause data loss, and is known to corrupt
   the Atmel AT24RF08 EEPROM found on many IBM Thinkpad laptops.
   See also the `i2cdetect man page
   <http://manpages.ubuntu.com/manpages/bionic/man8/i2cdetect.8.html>`_
 ```

**Review by dbkinder — CHANGES_REQUESTED** [review]
@nashif Aren't we worried about providing a sample that could knowingly corrupt system data?

**Review by dbkinder — APPROVED** [review]
+1 for doc changes, thanks.  I'm still a bit concerned about having a sample that knowingly can corrupt a system, but I'll leave that discussion for the developers to resolve.

**Review by carlescufi — APPROVED** [review]

**Pull request merged**